streams/src/main/java/org/apache/kafka/streams/processor/StateRestoreListener.java

```diff
@@ -29,9 +29,12 @@
  * Users desiring stateful operations will need to provide synchronization internally in
  * the {@code StateRestorerListener} implementation.
  *
- * When used for monitoring a single {@link StateStore} using either {@link AbstractNotifyingRestoreCallback} or
- * {@link AbstractNotifyingBatchingRestoreCallback} no synchronization is necessary
- * as each StreamThread has its own StateStore instance.
+ * Note that this listener is only registered at the per-client level and users can base on the {@code storeName}
```

Comment:
Updated the javadoc to emphasize one should not compound the two interfaces, cc @vvcephei .

Reply:
Thanks!